modules/UI/videolayout/VideoContainer.js

```diff
@@ -102,13 +99,22 @@ function getCameraVideoSize(videoWidth,
             availableWidth = availableHeight * aspectRatio;
         }
 
-        if (availableHeight * aspectRatio < videoSpaceWidth) {
+        if (aspectRatio <= 1) {
```

Comment:
You're right about the confusion here. Looks like it's trying to make sure the height of the video can fit the height of the screen no matter what. It makes me wonder then if MAXIMUM_ZOOMING_COEFFICIENT is needed at all.